sympy/geometry/point.py

```diff
@@ -261,59 +296,127 @@ def affine_rank(*points):
         if b.is_zero:
             raise ValueError("Cannot project to the zero vector.")
-        if a.ambient_dimension != b.ambient_dimension:
-            raise ValueError("Points must be of the same dimension")
 
         return b*(a.dot(b) / b.dot(b))
 
     @staticmethod
-    def pointify(a, dimension=None, ensure_geometry_entity=True, ensure_point=True, **kwargs):
-        """If `a` is a `Point` or a non-iterable, it is left alone.
-        Otherwise, it is converted to a point with `kwargs` passed along
-        to `Point()`. If `dimension` is set (default `None`), a dimension check is carried out.
-        If `ensure_geometry_entity` is set (default `True`), a type check for GeometryEntity
-        is carried out and the return value will be a GeometryEntity.
-        If `ensure_point` is set (default `True`), a type check for Point is carried out
-        and the return value will be a Point."""
+    def pointify(a, dimension=None, strict=False, ensure_point=True):
+        """Return a Point with coordinates given by `a` if
+        `ensure_point` is True or if `a` was already a Point else return
+        a Tuple if `ensure_point` is False and `a` is a sequence. When
+        `dimension` is given the returned value will have a matching
+        length with padding or trimming of zeros as necessary
+        (default) unless `strict` is True.
 
+
+        Parameters
+        ==========
+
+        a: Point or sequence of coordinates
+        dimension: int
+        strict: bool
+        ensure_point: bool
+
+        Raises
+        ======
+
+        ValueError
+            If `strict` is True and there is a dimension
+            mismatch or if a non-zero dimension is encountered
+            beyond the given `dimension`.
+
+        TypeError
+            If `a` is not a GeometryEntity and cannot be made into a
+            Point.
+
+        Examples
+        ========
+
+        >>> from sympy import Point
+        >>> p = Point.pointify
+        >>> p([1, 2])
+        Point2D(1, 2)
+        >>> p(Point(1, 2), dimension=3)
+        Point3D(1, 2, 0)
+
+        Zeros will be padded ot trimmed to meet the requested
+        dimensions:
+
+        >>> p(Point(1, 2, 0), dimension=2)
+        Point2D(1, 2)
+        >>> p(Point(1, 2), dimension=3)
+        Point3D(1, 2, 0)
+
+        But a nonzero dimension will not be removed:
+
+        >>> p(Point(1, 2, 1), dimension=2)
+        Traceback (most recent call last):
+        ...
+        ValueError: too many nonzero dimensions
+
+        To enforce a given number of dimensions, set
+        strict to True:
+
+        >>> p(Point(1, 2, 0), dimension=2, strict=True)
+        Traceback (most recent call last):
+        ...
+        ValueError: Point has wrong number of dimensions
+
+        >>> p((1, 2, 0), dimension=4, strict=True)
+        Traceback (most recent call last):
+        ...
+        ValueError: wrong number of dimensions
+        """
```

Comment:
I was just following the example of whoever went through and highly upgraded the geometry docstrings some time ago. :-)